### PR TITLE
New messages for cookie SameSite attribute in WebContainer

### PR DESCRIPTION
**Pull request by pmd1nh**
messages for #10788

### CODE REVIEW

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 1]
```diff
@@ -1603,3 +1603,18 @@ exception.occurred.while.running.ServletContainerInitializers.onStartup.useracti
...
 no.servlet.defined.for.servlet.filter.name.mapping.useraction=Define the servlet which is used in the servlet name filter mapping.
+
+# {0} Cookie name that has conflict SameSite attribute
+# {1} current SameSite attribute value
+# {2} new SameSite attribute value
+duplicate.samesite.attribute.for.cookie.name=CWWWC0003W: A cookie name [{0}] with SameSite attribute value [{1}] is configured with new value [{2}].  No SameSite attribute will be added for this cookie.
+duplicate.samesite.attribute.for.cookie.name.explanation=A cookie name must not have multiple SameSite attribute values.  The attribute will not be added for the cookie.
+duplicate.samesite.attribute.for.cookie.name.useraction=The deployment descriptor's context parameters cookie names and values for 'SameSiteCookies_Lax', 'SameSiteCookies_None', and 'SameSiteCookies_Strict' must be unique across all three lists. 
+
+duplicate.reported.for.samesite.attribute.cookie.name=CWWWC0004W: A cookie name [{0}] with duplicate SameSite attribute value [{1}] was found previously.  No SameSite attribute will be added for this cookie.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: **change**
A cookie name [{0}] with duplicate SameSite attribute value [{1}] was found previously.  No SameSite attribute will be added for this cookie.

**to**
The {1} SameSite attribute value  already exists for the {0} cookie name.  No SameSite attribute is added for this cookie.

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 2]
```diff
@@ -1603,3 +1603,18 @@ exception.occurred.while.running.ServletContainerInitializers.onStartup.useracti
...
+duplicate.reported.for.samesite.attribute.cookie.name.explanation=A cookie name must not have multiple SameSite attribute values.  The attribute will not be added for the cookie.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: **change**
A cookie name must not have multiple SameSite attribute values.  The attribute will not be added for the cookie.

**to**
A cookie name cannot have multiple SameSite attribute values.  The attribute is not added for the cookie.

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 3]
```diff
@@ -1603,3 +1603,18 @@ exception.occurred.while.running.ServletContainerInitializers.onStartup.useracti
...
+duplicate.reported.for.samesite.attribute.cookie.name.useraction=The deployment descriptor's context parameters cookie names and values for 'SameSiteCookies_Lax', 'SameSiteCookies_None', and 'SameSiteCookies_Strict' must be unique across all three lists. 
+
+too.many.wildcard.configured.for.samesite.cookie.name=CWWWC0005W: Too many wildcard cookie name configured for SameSite attribute.  No attribute will be added for any cookie.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: **change**
Too many wildcard cookie name configured for SameSite attribute.  No attribute will be added for any cookie.

**to**
Multiple wildcard cookie names are configured for SameSite attributes.  No attribute is added for any cookie.

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 4]
```diff
@@ -1603,3 +1603,18 @@ exception.occurred.while.running.ServletContainerInitializers.onStartup.useracti
...
+too.many.wildcard.configured.for.samesite.cookie.name.explanation=Too many wildcard cookie names are configured for multiple SameSite attributes. You can configure only one wildcard cookie name across all SameSite attributes.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: **change**
Too many wildcard cookie names are configured for multiple SameSite attributes. You can configure only one wildcard cookie name across all SameSite attributes.

**to**
Multiple wildcard cookie names are configured for one or more SameSite attributes. You can configure only one wildcard cookie name across all SameSite attributes.

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 5]
```diff
@@ -1603,3 +1603,18 @@ exception.occurred.while.running.ServletContainerInitializers.onStartup.useracti
 no.servlet.defined.for.servlet.filter.name.mapping=CWWWC0002W: No servlet definition is found for the {0} servlet name in the {1} filter mapping.
 no.servlet.defined.for.servlet.filter.name.mapping.explanation=The servlet name is used in filter mapping, but its definition has not been defined.
 no.servlet.defined.for.servlet.filter.name.mapping.useraction=Define the servlet which is used in the servlet name filter mapping.
+
+# {0} Cookie name that has conflict SameSite attribute
+# {1} current SameSite attribute value
+# {2} new SameSite attribute value
+duplicate.samesite.attribute.for.cookie.name=CWWWC0003W: A cookie name [{0}] with SameSite attribute value [{1}] is configured with new value [{2}].  No SameSite attribute will be added for this cookie.
+duplicate.samesite.attribute.for.cookie.name.explanation=A cookie name must not have multiple SameSite attribute values.  The attribute will not be added for the cookie.
+duplicate.samesite.attribute.for.cookie.name.useraction=The deployment descriptor's context parameters cookie names and values for 'SameSiteCookies_Lax', 'SameSiteCookies_None', and 'SameSiteCookies_Strict' must be unique across all three lists.
```
[the remark this answers is not in the document]
Comment (pmd1nh) [reply dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: SameSiteCookies_Lax, SameSiteCookies_None and SameSiteCookies_Strict are the context parameters setting in the deployment descriptor web.xml

The value in this case is actually the cookie name itself.  I changed the useraction to:

duplicate.samesite.attribute.for.cookie.name.useraction=The cookie name set in the deployment descriptor's context parameters for 'SameSiteCookies_Lax', 'SameSiteCookies_None', and 'SameSiteCookies_Strict' must be unique across all three lists.

**Comment by pmd1nh** [comment]
@dmuelle I pushed new changes up.  Thanks for review!

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 6]
```diff
@@ -1603,3 +1603,18 @@ exception.occurred.while.running.ServletContainerInitializers.onStartup.useracti
 no.servlet.defined.for.servlet.filter.name.mapping=CWWWC0002W: No servlet definition is found for the {0} servlet name in the {1} filter mapping.
 no.servlet.defined.for.servlet.filter.name.mapping.explanation=The servlet name is used in filter mapping, but its definition has not been defined.
 no.servlet.defined.for.servlet.filter.name.mapping.useraction=Define the servlet which is used in the servlet name filter mapping.
+
+# {0} Cookie name that has conflict SameSite attribute
+# {1} current SameSite attribute value
+# {2} new SameSite attribute value
+duplicate.samesite.attribute.for.cookie.name=CWWWC0003W: A {0} cookie name with the existing SameSite attribute value {1} is configured with the {2} new value. No SameSite attribute is added for this cookie.
+duplicate.samesite.attribute.for.cookie.name.explanation=A cookie name cannot have multiple SameSite attribute values. The attribute is not added for the cookie.
+duplicate.samesite.attribute.for.cookie.name.useraction=The deployment descriptor's context parameters cookie names for 'SameSiteCookies_Lax', 'SameSiteCookies_None', and 'SameSiteCookies_Strict' must be unique across all three lists.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: **change**
The deployment descriptor's context parameters cookie names for 'SameSiteCookies_Lax', 'SameSiteCookies_None', and 'SameSiteCookies_Strict' must be unique across all three lists. 

**to**
Review the deployment descriptor configuration and make sure the cookie names that are set in the context parameters do not have more than one SameSite variable value.

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 7]
```diff
@@ -1603,3 +1603,18 @@ exception.occurred.while.running.ServletContainerInitializers.onStartup.useracti
...
+duplicate.samesite.attribute.for.cookie.name.explanation=A cookie name cannot have multiple SameSite attribute values. The attribute is not added for the cookie.
+duplicate.samesite.attribute.for.cookie.name.useraction=The deployment descriptor's context parameters cookie names for 'SameSiteCookies_Lax', 'SameSiteCookies_None', and 'SameSiteCookies_Strict' must be unique across all three lists. 
+
+duplicate.reported.for.samesite.attribute.cookie.name=CWWWC0004W: The {1} SameSite attribute value already exists for the {0} cookie name. No SameSite attribute is added for this cookie.
+duplicate.reported.for.samesite.attribute.cookie.name.explanation=A cookie name cannot have multiple SameSite attribute values. The attribute is not added for the cookie.
+duplicate.reported.for.samesite.attribute.cookie.name.useraction=The cookie name set in the deployment descriptor's context parameters for 'SameSiteCookies_Lax', 'SameSiteCookies_None', and 'SameSiteCookies_Strict' must be unique across all three lists.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: **change**
The cookie name set in the deployment descriptor's context parameters for 'SameSiteCookies_Lax', 'SameSiteCookies_None', and 'SameSiteCookies_Strict' must be unique across all three lists.

**to**
Review the deployment descriptor configuration and make sure the cookie names that are set in the context parameters do not have more than one SameSite variable value.

Comment (pmd1nh) [reply dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: @dmuelle 
hi David,

This is an example of what the descriptor's context param config looks like

<context-param>
            <param-name>SameSiteCookies_Lax</param-name>
            <param-value>**WebContainerSS**,LAX1,WSSLAX2,WSSLAX_3</param-value>
    </context-param>

<context-param>
            <param-name>SameSiteCookies_None</param-name>
            <param-value>**WebContainerSS**,NONE1,WSSNONE2, WSSNONE_3</param-value>
    </context-param>

The user can't put the same <param-value>, which is the cookie name, in more than one <param-name>, which is the SameSite Attribute, section.  In the above example, **WebContainerSS** will get a WARNING.

Comment (pmd1nh) [reply dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: communicated with David via Slack as the bracket code snippet does not work in this comment section.

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops (hunk 6 above)
Comment (dmuelle) [reply dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: edited above comment

**Comment by skasund** [comment]
L2 message review completed

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 8]
```diff
@@ -1603,3 +1603,18 @@ exception.occurred.while.running.ServletContainerInitializers.onStartup.useracti
...
 no.servlet.defined.for.servlet.filter.name.mapping.useraction=Define the servlet which is used in the servlet name filter mapping.
+
+# {0} Cookie name that has conflict SameSite attribute
+# {1} current SameSite attribute value
+# {2} new SameSite attribute value
+duplicate.samesite.attribute.for.cookie.name=CWWWC0003W: A {0} cookie name with the existing SameSite attribute value {1} is configured with the {2} new value. No SameSite attribute is added for this cookie.
+duplicate.samesite.attribute.for.cookie.name.explanation=A cookie name cannot have multiple SameSite attribute values. The attribute is not added for the cookie.
+duplicate.samesite.attribute.for.cookie.name.useraction=The deployment descriptor's context parameters cookie names for 'SameSiteCookies_Lax', 'SameSiteCookies_None', and 'SameSiteCookies_Strict' must be unique across all three lists. 
+
+duplicate.reported.for.samesite.attribute.cookie.name=CWWWC0004W: The {1} SameSite attribute value already exists for the {0} cookie name. No SameSite attribute is added for this cookie.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: **change**
The {1} SameSite attribute value already exists for the {0} cookie name.
**to**
The {0} cookie name that is configured for the {1} SameSite attribute value was previously identified as a duplicate. No SameSite attribute is added for this cookie.

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops [hunk 9]
```diff
@@ -1603,3 +1603,18 @@ exception.occurred.while.running.ServletContainerInitializers.onStartup.useracti
...
+duplicate.reported.for.samesite.attribute.cookie.name.explanation=A cookie name cannot have multiple SameSite attribute values. The attribute is not added for the cookie.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: **change**
A cookie name cannot have multiple SameSite attribute values. The attribute is not added for the cookie.
**to**
A cookie name cannot be configured for multiple SameSite attribute values. This cookie name has already been identified as a duplicate.

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops (hunk 6 above)
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: **change**
The deployment descriptor's context parameters cookie names for 'SameSiteCookies_Lax', 'SameSiteCookies_None', and 'SameSiteCookies_Strict' must be unique across all three lists. 
**to**
Check the deployment descriptor configuration to make sure the cookie name is configured for only one of the following SameSite attribute values: 'SameSiteCookies_Lax', 'SameSiteCookies_None', or 'SameSiteCookies_Strict'.

File: dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops (hunk 7 above)
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/com/ibm/ws/webcontainer/resources/Messages.nlsprops]: **change**
The cookie name set in the deployment descriptor's context parameters for 'SameSiteCookies_Lax', 'SameSiteCookies_None', and 'SameSiteCookies_Strict' must be unique across all three lists.
**to**
Check the deployment descriptor configuration to make sure the cookie name is configured for only one of the following SameSite attribute values: 'SameSiteCookies_Lax', 'SameSiteCookies_None', or 'SameSiteCookies_Strict'.

**Comment by pmd1nh** [comment]
Updated Copyright after David final approval

 Copyright (c) 1997, 2020 IBM Corporation and others.

**Comment by pmd1nh** [comment]
#run-libby-bot

**Pull request merged**